### ISSUE TITLE
Annotations for false positives: MissingUnpackerDep

### ISSUE THREAD

**Issue by robbat2** [pkgcore/pkgcheck#214]
I have an ebuild that directly consumes some zipfiles without using `unzip`. This triggers `MissingUnpackerDep BDEPEND="app-arch/unzip"` and there is no way to annotate it that it's really not missing!

**Comment by robbat2** [comment]
CC @mgorny

**Comment by mgorny** [comment]
CC @ulm

**Comment by mgorny** [comment]
Since we have annotations already planned for EAPI 8, I'm wondering if it's worth considering to add another one indicating 'do not unpack this distfile'. That'd kill two birds with one stone, also preventing `default_src_unpack` from unpacking it.

**Comment by robbat2** [comment]
That idea would be even better, for the ebuilds that directly consume distfiles for various reasons, and explicitly do not want them unpacked.

**Comment by ulm** [comment]
I wonder if SRC_URI is the right place for this. Normally ebuilds want to unpack all files in ${A}. Anything special (like no unpacking, or nested unpacking) can be handled programmatically in src_unpack. (And I guess pkgcheck would miss an unzip that is only needed for unpacking of an inner archive?)

This is different from `mirror` or `fetch` annotations, where there is currently no way to specify the behaviour.

Maybe another QA variable (like `QA_NO_UNPACK`) would be a more flexible solution, which would not require an EAPI bump?

**Comment by mgorny** [comment]
Not sure if I hadn't mentioned it somewhere earlier but another option is to disable this check if the ebuild is defining a custom `src_unpack()`.

**Comment by radhermit** [comment]
I don't think this is worth approaching or working around until something like the SRC_URI annotations mentioned above exist in some fashion.
